Packs/XSOARmirroring/Integrations/XSOARmirroring/XSOARmirroring.py

```diff
@@ -828,34 +824,35 @@ def get_and_dedup_incidents(client: Client, last_fetched_incidents: list[dict],
     Returns:
         tuple[list[dict], list[str]: The list of dedup incidents and ID list of incidents of current fetch.
     """
-    last_fatched_incident = (dateparser.parse(last_fetched_incidents[-1]["created"])
-                             if last_fetched_incidents else None)
+    last_fatched_incident = dateparser.parse(str(last_fetch))
 
     new_incidents: list = []
     page = 0
     while len(new_incidents) < max_results:
         incidents = client.search_incidents(
-        query=query,
-        max_results=max_results,
-        start_time=last_fetch,
-        field="created",
-        page=page,
+            query=query,
+            max_results=max_results,
+            start_time=last_fetch,
+            field="created",
+            page=page,
         )
-        # Case 1: Empty response len(incidents) == 0
+        # Case 1: Empty response.
         if len(incidents) == 0:
             break
         for incident in incidents:
             if len(new_incidents) >= max_results:
                 break
-            incident_dict = {"id": incident["id"], "created": incident["created"]}
-            incident_creation_time = dateparser.parse(incident["created"])
-            if incident_dict not in last_fetched_incidents:
+            incident_id = incident.get("id")
+            incident_creation_time = dateparser.parse(incident.get("created", datetime.now()))
```

Comment:
Is there a chance that the created field won't be present? Im not sure that setting the datetime now as the creation time is the correct approach